xpdan/db_utils.py

```diff
@@ -0,0 +1,100 @@
+from pprint import pprint
+from .utils import _timestampstr
+
+
+def need_name_here(hdrs, key, verbose=True):
+    """Build a dictionary where the keys are the unique values for the given
+    key and the values are the positions of the headers for that key
+
+    Parameters
+    ----------
+    hdrs: list of Header objects
+        The headers to be searched
+    key: str
+        The key to be searched for
+    verbose: bool, optional
+        If true prints the results. Defaults to True
+
+    Returns
+    -------
+    dict:
+        The dictionary of unique values and positions
+    """
+    d = {}
+    for i, hdr in enumerate(hdrs):
+        if key in hdr.start.keys():
+            if hdr.start[key] in d.keys():
+                d[hdr.start[key]].append(i)
+            else:
+                d[hdr.start[key]] = [i]
+    if verbose:
+        pprint(d)
+    return d
+
+
+def scan_diff(hdrs, verbose=True):
+    """Get the metadata differences between scans
```

Comment:
add '...between scans in hdrs list'

This function prints all the header information from each header in the hdrs list, except for any key:value pairs that have the same value for all the headers in the list.